src/main/java/com/crowsofwar/avatar/common/entity/data/Behavior.java

```diff
@@ -96,7 +109,7 @@ public int getId() {
 	 * Called every update tick.
 	 *
 	 * @return Next Behavior. Return <code>this</code> to continue the Behavior.
-	 * Never return null.
+	 *         Never return null.
 	 */
 	public abstract Behavior onUpdate(E entity);
```

Comment:
```suggestion
	public abstract Behavior<? extends E> onUpdate(E entity);
```
Would make the cast above not necessary anymore

Reply:
I guess I'll do that later for all entities